core/modules/xrdsvc/SsiRequest.cc

```diff
@@ -436,6 +436,18 @@ bool SsiRequest::replyStream(StreamBuffer::Ptr const& sBuf, bool last) {
 }
 
 
+bool SsiRequest::sendMetadata(const char *buf, int blen) {
+    Status stat = SetMetadata(buf, blen);
+    if (stat == XrdSsiResponder::wasPosted) return true;
+    if (stat == XrdSsiResponder::notActive) {
+        LOGS(_log, LOG_LVL_ERROR, "failed to setMetadata notActive");
+    } else {
+        LOGS(_log, LOG_LVL_ERROR, "failed to setMetadata notPosted blen=" << blen);
+    }
+    return false;
```

Comment:
Well, this will work as long as a new return code is not introduced. While that is highly unlikely, programming for the case would be a good conservative approach.

Reply:
Changed to a switch statement.